library/common/glfw/embedder.cc

```diff
@@ -385,6 +398,37 @@ FlutterWindowRef FlutterEmbedderCreateWindow(
   return state;
 }
 
+FlutterEngine g_background_engine = nullptr;
+
+static void SigHupHandler(int dummy) {
```

Comment:
The signal handler aspect of this seems like something that belongs in the embedder client rather than in the framework. Having the infinite loop in the windowed version is there because of the need to poll Flutter and drive UI events, but that's not an issue here. I would suggest that starting the engine should return an opaque handle like the FlutterWindowRef, and then the API surface can have a shutdown method that takes that handle. That way an embedder can decide to run forever, have a SigHup handler, or control lifetime however else they want to.

Reply:
Yeah, sent for review to early. The appveyor error made me double check this and agree that it's kinda weird to have it here. Fixing.